### ISSUE TITLE
The addition of CAPTCHA to the registration process

### ISSUE THREAD

**Issue by NHSBSA-Open-Data** [nhsbsa-data-analytics/open-data-roadmap#60]
## Overview
The removal of registration via API and the addition of CAPTCHA to the registration process is to reduce the automatic registering of accounts via online 'bots'.

## What is affected?
The registration process can currently be accessed and completed via API through the use of a 'bot'

## Timeframe of solution
End of July

## Solution Details
1. CKAN allow the user to register to the portal via the API using /api/3/action/user_create endpoint and people can exploit the API registration process and create spam users if they use that.
We have stopped the registration via API using https://docs.ckan.org/en/2.8/maintaining/configuration.html#ckan-auth-create-user-via-api
So user can register via UI https://ckan.nhs.staging.datopian.com/user/register but cannot register via API.

### What went wrong?
In recent days we have faced issue of bot registration which might be using UI Automation for the registration of the user. So we need to add CAPTCHA to the user registration form in CKAN.

### Why did it happen?
As part of the registration in CKAN user can either register via API or register via UI, the registration via API is not a problem as it is already blocked. The problem is with the registration via UI.

### The solution implemented
add CAPTCHA to the user registration form in CKAN. 

### How are we ensuring this does not happen again
One key reason for this issue is the absence of CAPTCHA or other verification processes during registration. Enabling CAPTCHA would make it significantly more difficult for automated systems to exploit the registration process. Although we have blocked the offending IP addresses, this measure alone is insufficient to fully resolve the problem.

**Comment by NHSBSA-Open-Data** [comment]
@shubham-mahajan Please can you add further details to the above ticket before it is released?

**Comment by shubham-mahajan** [comment]
@NHSBSA-Open-Data 

As part of the registration in CKAN user can either register via API or register via UI, the registration via API is not a problem as it is already blocked. The problem is with the registration via UI, so we need to add CAPTCHA to the user registration form in CKAN. In recent days we have faced issue of bot registration which might be using UI Automation for the registration of the user. So we need to add CAPTCHA to the user registration form in CKAN.

One key reason for this issue is the absence of CAPTCHA or other verification processes during registration. Enabling CAPTCHA would make it significantly more difficult for automated systems to exploit the registration process. Although we have blocked the offending IP addresses, this measure alone is insufficient to fully resolve the problem.
